src/presentation/components/FormLoaderStatus/index.spec.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { RenderResult, render } from '@testing-library/react';
+import { FormLoaderStatus } from '@/presentation/components';
+import Context from '@/presentation/hooks/form';
+
+const renderComponent = (isLoading: boolean, mainError: string): RenderResult => {
+  return render(
+    <Context.Provider value={{ state: { isLoading, mainError } }}>
+      <FormLoaderStatus />
+    </Context.Provider>
+  );
+};
+
+describe('FormLoaderStatus Component', () => {
+  test('Renders spinner when isLoading is true', () => {
+    const isLoading = true;
+    const mainError = '';
+    const { getByTestId } = renderComponent(isLoading, mainError);
+    const spinner = getByTestId('spinner');
+    expect(spinner).toBeInTheDocument();
+  });
+
+  test('Does not render spinner when isLoading is false', () => {
+    const isLoading = false;
+    const mainError = '';
+    const { queryByTestId } = renderComponent(isLoading, mainError);
```

Comment:
you can attribute an empty string as default to mainError in line 6, so you dont need to declare in every test, and I recommend you to use getByTestId as the other scenarios

Reply:
empty string part is done. 

I can use `getByTestId` but it throws error when the element is not rendered but on other side `queryByTestId` doesn't throw error. Please confirm me once if i have to change ?